eng/common/scripts/Update-DocsMsPackages2.ps1

```diff
@@ -0,0 +1,129 @@
+<#
+.SYNOPSIS
+Update docs.microsoft.com CI configuration with provided metadata
+
+.DESCRIPTION
+Update docs.microsoft.com CI configuration with metadata in the docs.microsoft.com repo's
+/metadata folder. The docs.microsoft.com repo's /metadata folder allows onboarding of
+packages which have not released to a central package manager.
+
+* Onboard new packages, update existing tracked packages, leave other packages
+  in place. (This is implemented on a per-language basis by
+  $UpdateDocsMsPackagesFn)
+
+.PARAMETER DocRepoLocation
+Location of the docs.microsoft.com reference docs repo.
+
+.PARAMETER PackageSourceOverride
+Optional parameter to supply a different package source (useful for daily dev
+docs generation from pacakges which are not published to the default feed). This
+variable is meant to be used in the domain-specific business logic in
+&$UpdateDocsMsPackagesFn
+
+.PARAMETER ImageId
+Optional The docker image for package validation in format of '$containerRegistry/$imageName:$tag'.
+e.g. azuresdkimages.azurecr.io/jsrefautocr:latest
+
+#>
+param (
+  [Parameter(Mandatory = $true)]
+  [string] $DocRepoLocation, # the location of the cloned doc repo
+
+  [Parameter(Mandatory = $false)]
+  [string] $PackageSourceOverride,
+
+  [Parameter(Mandatory = $false)]
+  [string] $ImageId
+)
+
+. (Join-Path $PSScriptRoot common.ps1)
+. "$PSScriptRoot/../../scripts/docs/Docs-Onboarding.ps1"
+
+Set-StrictMode -Version 3
+
+function GetMetadata($moniker) {
+  $jsonFiles = Get-ChildItem -Path (Join-Path $DocRepoLocation "metadata/$moniker") -Filter *.json
+  $metadata = @()
+  foreach ($jsonFile in $jsonFiles) {
+    # Converting to a hashtable gives more beneficial semantics for handling
+    # metadata (easier to check existence of property, easier to add new
+    # properties)
+    $metadata += Get-Content $jsonFile -Raw | ConvertFrom-Json -AsHashtable
+  }
+
+  return $metadata
+}
+
+function ValidatePackageForOnboarding2($package) {
```

Comment:
I don't like 2's :)

Reply:
I like them a lot less than you dislike them so I'm very happy to do things differently. 

This particular function name was named something like `ValidatePackage` but it collided with a name in `Language-Settings.ps1` and so my thinking here was to write a function which was sufficiently unique that I didn't need to go check for it in the other languages' `Language-Settings.ps1` files.

Reply:
Is this function scoped to this script? If so probably safe to remove the 2 still.